tests/cpp/operator/batchnorm_test.cc

```diff
@@ -673,15 +671,24 @@ testForwardAndBackward(const bool isGPU,
     cycleCount);
 }
 
+// NOTE: This should know which version to use (V1, mkl, etc)
+struct BatchNormCoreOpProp : public mxnet::test::op::CoreOpProp {
+  const mxnet::op::BatchNormParam& getParam() const {
+    CHECK(false); // Not implemented
+    static mxnet::op::BatchNormParam dummy;
+    return dummy;
+  }
+};
+
 template<typename OperatorExecutor>
-static test::op::OpInfoPair<mxnet::op::BatchNormV1Prop, mxnet::op::BatchNormProp, OperatorExecutor>
+static test::op::OpInfoPair<BatchNormCoreOpProp, BatchNormCoreOpProp, OperatorExecutor>
```

Comment:
This is where the version of the batch norm operator used to be specified, by the OperatorProp derivative.  However it's done in your API would tend to be specified BatchNormCoreOp or its derivative.